oss-internship-2020/libuv/examples/helloworld.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <linux/futex.h>
+#include <syscall.h>
+#include <uv.h>
+
+#include <iostream>
+
+#include "sandboxed_api/util/flag.h"
+#include "uv_sapi.sapi.h"
+
+class UVSapiHelloworldSandbox : public UVSandbox {
+ private:
+  std::unique_ptr<sandbox2::Policy> ModifyPolicy(
+      sandbox2::PolicyBuilder*) override {
+    return sandbox2::PolicyBuilder()
+        .AllowDynamicStartup()
+        .AllowExit()
+        .AllowFutexOp(FUTEX_WAKE_PRIVATE)
+        .AllowSyscalls({__NR_epoll_create1, __NR_eventfd2, __NR_pipe2})
+        .AllowWrite()
+        .BuildOrDie();
+  }
+};
+
+int main(int argc, char* argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+  google::InitGoogleLogging(argv[0]);
+
+  absl::Status status;
+
+  // Initialize sandbox2 and sapi
+  UVSapiHelloworldSandbox sandbox;
+  status = sandbox.Init();
+  if (!status.ok()) {
+    LOG(FATAL) << "Couldn't initialize Sandboxed API: " << status;
+  }
+  UVApi api(&sandbox);
```

Comment:
Regarding error handling. I commented on this in Ina's PFFFT sandbox: At Google we pervasively use `absl::Status` everywhere, often with the status macros.
You should refactor your error handling like this: https://github.com/google/sandboxed-api/blob/master/oss-internship-2020/pffft/main_pffft_sandboxed.cc#L88

That way, you don't have to have multiple `LOG(FATAL)` statements and it makes your code more readable, because the "happy path" is clearer.

Same for the other files.